wagtail/admin/views/editing_sessions.py

```diff
@@ -0,0 +1,85 @@
+from django.apps import apps
+from django.contrib.admin.utils import unquote
+from django.contrib.contenttypes.models import ContentType
+from django.http import Http404, JsonResponse
+from django.shortcuts import get_object_or_404
+from django.utils import timezone
+from django.views.decorators.http import require_POST
+
+from wagtail.admin.models import EditingSession
+from wagtail.admin.utils import get_user_display_name
+from wagtail.models import Page
+from wagtail.permissions import page_permission_policy
+
+
+def ping(request, app_label, model_name, object_id, session_id):
+    try:
+        model = apps.get_model(app_label, model_name)
+    except LookupError:
+        raise Http404
+
+    unquoted_object_id = unquote(object_id)
+
+    content_type = ContentType.objects.get_for_model(model)
+
+    obj = get_object_or_404(model, pk=unquoted_object_id)
+    if isinstance(obj, Page):
+        permission_policy = page_permission_policy
+    else:
+        try:
+            permission_policy = model.snippet_viewset.permission_policy
+        except AttributeError:
+            # model is neither a Page nor a snippet
+            raise Http404
+
+    if not permission_policy.user_has_permission_for_instance(
+        request.user, "change", obj
+    ):
+        raise Http404
+
+    try:
+        session = EditingSession.objects.get(
+            id=session_id,
+            user=request.user,
+            content_type=content_type,
+            object_id=unquoted_object_id,
+        )
+    except EditingSession.DoesNotExist:
+        session = EditingSession(
+            content_type=content_type,
+            object_id=unquoted_object_id,
+            user=request.user,
+        )
+
+    session.last_seen_at = timezone.now()
+    session.save()
+
+    other_sessions = (
+        EditingSession.objects.filter(
+            content_type=content_type,
+            object_id=unquoted_object_id,
+            last_seen_at__gte=timezone.now() - timezone.timedelta(minutes=1),
+        )
+        .exclude(id=session.id)
```

Comment:
Note: we're still treating "different sessions by the same user for the same object" as "other sessions" here – which is fine. I think we still want this behaviour later (for when you save changes in a different tab), but we _might_ want to hide those sessions if they don't have any changes,  or at least deprioritise them in the list. That's for later when we have the revision ID / "has unsaved changes" flag though.

And when we do explicitly handle those, we need to ensure we have tests for such cases. (Right now I don't think I see them, but it's OK as we don't treat it any different yet...)

Question: do we want to `order_by("-last_seen_at")` here? Or maybe add a default `ordering` on the `EditingSession.Meta`? We can always add it later when we add more logic to the list's prioritisation though.

Reply:
Yep, I think we can make decisions on both of those later.